shell/platform/common/cpp/accessibility_bridge.h

```diff
@@ -0,0 +1,240 @@
+  ///
+  /// The accessibility actions are generated by the native accessibility system
+  /// when users interacted with the assistive technologies. Those actions
+  /// needed to be sent to the Flutter framework.
+  class AccessibilityBridgeDelegate {
+   public:
+    virtual ~AccessibilityBridgeDelegate() = default;
+    //------------------------------------------------------------------------------
+    /// @brief      Handle accessibility events generated due to accessibility
+    /// tree
+    ///             changes.
+    ///
+    /// @param[in]  targeted_event      The object that contains both the
+    /// generated
+    ///                                 event and the event target.
+    /// @param[in]  bridge              The pointer to the accessibility bridge
+    /// that
+    ///                                 can be used for querying the
+    ///                                 accessibility information at the time
+    ///                                 the event is fired.
+    virtual void OnAccessibilityEvent(
+        AXEventGenerator::TargetedEvent targeted_event,
+        AccessibilityBridge* bridge) = 0;
+
+    //------------------------------------------------------------------------------
+    /// @brief      Dispatch accessibility action back to the Flutter framework
+    ///
+    /// @param[in]  target              The semantics node id of the action
+    /// target.
+    /// @param[in]  action              The generated flutter semantics action.
+    /// @param[in]  data                Additional data associated with the
+    /// action.
+    /// @param[in]  data_size           The length of the additional data.
+    virtual void DispatchAccessibilityAction(uint16_t target,
+                                             FlutterSemanticsAction action,
+                                             uint8_t* data,
+                                             size_t data_size) = 0;
+  };
+  //------------------------------------------------------------------------------
+  /// @brief      Creates a new instance of a accessibility bridge.
+  ///
+  /// @param[in]  user_data           A custom pointer to the data of your
+  ///                                 choice. This pointer can be retrieve later
+  ///                                 through GetUserData().
+  AccessibilityBridge(std::unique_ptr<AccessibilityBridgeDelegate> delegate,
+                      void* user_data);
+  ~AccessibilityBridge();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Adds a semantics node update to the pending semantics update.
+  ///             Calling this method alone will NOT update the semantics tree.
+  ///             To flush the pending updates, call the CommitUpdates().
+  ///
+  /// @param[in]  node           A pointer to the semantics node update.
+  void AddFlutterSemanticsNodeUpdate(const FlutterSemanticsNode* node);
```

Comment:
Raw pointer usage. This probably wants to be a shared_ptr.

Reply:
Docs should explain whether this adds children or not, and when you need to call this on a node.

Reply:
This pointer is coming from the embedder API, and the embedder controls its lifetime exclusively. It is actually quite hard to predict when the data will be destroyed. I am not sure whether we should use smart pointer in here?

Reply:
I keep it as is, unless we want to create a separate memory copy in the embedding, we cannot use smart pointer here

Reply:
Can you point me to the code int he embedding that uses this?

Reply:
embedder calls
https://github.com/chunhtai/engine/blob/84ded7cac7152a80bc33124cea8ab4474402732d/shell/platform/darwin/macos/framework/Source/FlutterEngine.mm#L212

and it will then get passed directly into
https://github.com/chunhtai/engine/blob/84ded7cac7152a80bc33124cea8ab4474402732d/shell/platform/darwin/macos/framework/Source/FlutterEngine.mm#L623